### ISSUE TITLE
Allow for manual typing in of values when creating a fan curve

### ISSUE THREAD

**Issue by trparky** [Rem0o/FanControl.Releases#1538]
I'd like to see the ability to manually type in temperature and percentage values when creating a custom fan curve.

**Comment by Rem0o** [comment]
You can already, sort of, but you have to create a point first somewhere in the graph.
![image](https://user-images.githubusercontent.com/14118956/218577234-80c365d0-2942-4d0c-83be-f0137d230bb0.png)

**Comment by trparky** [comment]
Yes, but then when you select a point and you don't keep your hand perfectly steady, you can move the point. Perhaps a section where you can see the temp points and the percentages in a list where you can also edit them by hand.

**Comment by Rem0o** [comment]
@trparky just to be clear, the text field pointed with red arrows, you can type in them.

**Comment by trparky** [comment]
Yes, but when you select a point on the graph to see what its settings are, it moves it.

**Comment by Rem0o** [comment]
You only need to hover your mouse over it.

**Comment by trparky** [comment]
But as I move the mouse over the green little (but larger) dot, I don't get what I inputted.

I put 40% and 60c, press Enter. Hover my mouse over that dot again and wait... do I get that exact number again? No. It fluctuates based upon where my mouse cursor is hoving over the background graph.

For instance, I put the setting at 50% fan at 42c.
![image](https://user-images.githubusercontent.com/32105035/218620380-513fc50b-b0a5-41d5-8978-75478ebbe614.png)
Great.

Now, let's hover my mouse over it. Where do I put my mouse cursor? Do I put it in the center? The left? The right? Upper right? There's no locking on of the cursor to let me know that later, I'm getting the right setting back. For instance...
![image](https://user-images.githubusercontent.com/32105035/218620727-e91b4781-6015-4f1c-8a50-a867e1ca8b25.png)

Now, if you had an option to be able to tell the program to lock in on the EXACT location of the green little dot, it would work. But as of right now, that's not the case.

**Comment by trparky** [comment]
When you have a fan curve setup and you come back five minutes later, I don't remember my prior settings when I go back to add another point on the graph. I need to be able to know if I set that at 50c or 52c. I'm trying to dial in that perfect fan curve here and your program isn't allowing me to do that.

**Comment by Rem0o** [comment]
>  I need to be able to know if I set that at 50c or 52c.

The overlay on the graph in parenthesis is your exact cursor location, _it should not be used to tell the exact location of the green dot_, that's not what the overlay is meant for.  

If you want to know the exact location of a point, if it's 50 or 52: 

1) hover your mouse over that point until it becomes bigger
2) Read the measurement at the "Selected point" fields. 

In your last screenshot, even though your mouse isn't exactly on the point, it doesn't matter, the correct point is highlighted, and the measurement at the bottom is correct and you can rely on it, in this case 42,50.

I could make it so that the overlay will show the nearest point once you come close to it with your cursor, but that would make it hard/confusing to add another point close to that location.

**Comment by trparky** [comment]
But I clicked on the green dot and it immediately moved!!! That's not good user interface design!
> I could make it so that the overlay will show the nearest point once you come close to it with your cursor, but that would make it hard/confusing to add another point close to that location.
Then make it an option to allow that to show.

Oh, and I found another bug. If you go and modify the JSON file manually (which I know how to do) in a text editor and you tell the program to load a configuration file, the changes you made in that file will not reflect inside the program unless you exit the program and reload it. But wait, I just told it to load a configuration file. It isn't doing that.

**Comment by Rem0o** [comment]
>  If you go and modify the JSON file manually (which I know how to do) in a text editor and you tell the program to load a configuration file, the changes you made in that file will not reflect inside the program unless you exit the program and reload it

That's not exactly right. If you edit a file that is currently in use (loaded), then the program will think the file needs saving, since the state of the application no longer matches the config file, since you just changed it.  I should probably add a lock on the currently used json file to block that behavior in the first place.

Now if you edit any other json config file that is not the currently loaded config, then load that, you'll get your manual changes.

**Comment by trparky** [comment]
I was finally able to get your program to give me back the exact setting for the data point on the graph. However, I had to get that cursor dead center in the middle of the green dot.

Clicking on the dot shouldn't be able to move or edit it. Perhaps double-clicking or pressing the Shift key on the keyboard while clicking on it should activate the ability to move it.

**Comment by NeverUsedID** [comment]
What about an advanced view like that? Where you can see all points and add/remove/edit them in a small list view.

![grafik](https://user-images.githubusercontent.com/7545210/218747389-3717d234-86c7-454e-9c93-bf11901dc3bf.png)

**Comment by trparky** [comment]
Advanced view? Where?

**Comment by NeverUsedID** [comment]
> Advanced view? Where?

Sry, that was just an enhancment suggestion i created with Paint, to give the option to show all points and edit them directly. This "advanced view" is actually not there.

**Comment by trparky** [comment]
YES!!! That would be very nice to be added to the program!

**Comment by areasudon** [comment]
The good thing is I can have zero rpm mode on the cpu. You just need fan control and ignore CPU fan speed in the  asus bios for example.

**Comment by Rem0o** [comment]
https://github.com/Rem0o/FanControl.Releases/releases/tag/V147

>Clicking on the dot shouldn't be able to move or edit it.

When clicking onto a point, the point will no longer move
